src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -400,7 +477,20 @@ Status GcsActorManager::RegisterActor(
   // Mark the callback as pending and invoke it after the actor has been successfully
   // created.
   actor_to_register_callbacks_[actor_id].emplace_back(std::move(callback));
-  RAY_CHECK(registered_actors_.emplace(actor->GetActorID(), actor).second);
+
+  const auto &owner_address = actor->GetOwnerAddress();
+  auto node_id = ClientID::FromBinary(owner_address.raylet_id());
+  auto worker_id = WorkerID::FromBinary(owner_address.worker_id());
+  auto it = unresolved_actors_.find(node_id);
+  if (it != unresolved_actors_.end() && it->second.count(worker_id)) {
```

Comment:
Shouldn't the actor always be in unresolved_actors_ and registered_actors_ already? I don't understand why we need the if checks.

Reply:
> Shouldn't the actor always be in unresolved_actors_ and registered_actors_ already? I don't understand why we need the if checks.

Yes, u are right, I will fix it.